dagshub/data_engine/model/datasource.py

```diff
@@ -120,6 +130,20 @@ def all(self) -> "QueryResult":
         self._check_preprocess()
         return self._source.client.get_datapoints(self)
 
+    def select(self, selected: List[Field]):
+        self._select = []
+
+        for s in selected:
+            new_select_field = {"name": s.column}
+            if s.as_of_timestamp:
+                new_select_field["asOf"] = int(s.as_of_timestamp)
+            if s.alias:
+                new_select_field["alias"] = s.alias
```

Comment:
Turn this into a `to_dict()` function on the Field object
Then this whole function can be turned into
```
self._select = [s.to_dict() for s in selected]
```